### ISSUE TITLE
[P1] EncodedUnit.sse reports balanced-coordinate error when diagonals are present

### ISSUE THREAD

**Issue by RobTand** [RobTand/tessera#230]
Reviewed ea72d23cbf26a453d0d740afac537fb6d6b8afbc. EncodedUnit.sse is documented at encode.py:151 and :2846-2848 as unweighted squared reconstruction error in the source weight's own units. With fitted or supplied diagonals, :2332 replaces the source coordinates by work = Dv^-1 Wrot Du^-1. The final diagnostic at :2865 measures ||work - U*scale||² without restoring Dv and Du.

For balanced error E, source-weight SSE is ||Dv E Du||², since the remaining input rotation is orthogonal. Thus this diagnostic depends on the balancing gauge and is not the reconstruction error its name and contract promise. It can misprice diagnostic comparisons involving diagonals. No artifact byte depends on this field, so this finding does not itself change served values.

A cheap CPU witness uses seed-1 16x32 weights scaled by .02, E2M1 rate 3, one refit, supplied sv=2 and su=1:

```python
import torch
from tessera.alphabet import E2M1_GRID, build_forest
from tessera.encode import encode_unit
from tessera.decode import reconstruct_unit
from tessera.diagonals import Diagonals
from tessera.trellis import ConvCode

weight = torch.randn(16, 32, generator=torch.Generator().manual_seed(1)) * .02
forest = build_forest(3, grid=E2M1_GRID)
code = ConvCode(memory=6)
diagonals = Diagonals(torch.full((16,), 2., dtype=torch.float16),
                      torch.ones(32, dtype=torch.float16))
unit = encode_unit(weight, forest, (3,) * 32, code,
                   diagonals=diagonals, scale_refit=1)
source_sse = ((weight - reconstruct_unit(unit, forest, code)) ** 2).sum().item()
print(unit.sse, source_sse, source_sse / unit.sse)
# 0.0012608773540705442, 0.005043509416282177, 4.0
```

Run from the reviewed checkout with PYTHONPATH=src and native math threads at one. This was reproduced with torch 2.10.0+cpu, without GPU/model validation.

tests/test_math_audit_scale_and_trellis.py:278-323 tests the SSE contract only without diagonals; its reconstruction helper also omits restoring diagonals and rotation. tests/test_diagonals.py does not check sse.

Fix direction: compute the diagnostic from the complete stored reconstruction against the preserved original source, or apply both stored diagonal factors to the balanced error before taking its norm. Preserve the original source explicitly: encode_unit later reuses the variable named weights for branch weights. Acceptance should check supplied and fitted nonuniform diagonals, with and without rotation, and after release/refit, against reconstruct_unit rather than a helper that repeats the implementation's omissions.

No fix was attempted because the user explicitly requested an audit and issues without implementation changes.

Disposition: filed for a separate implementation agent. The user explicitly requested an audit and issues, with no fixes; no implementation changes were made for this finding.
